release/branch/cloudbuild.yaml

```diff
@@ -13,73 +13,214 @@
+    dir: 'kpt'
     volumes:
       - name: home
         path: /root
-  - name: 'mirror.gcr.io/library/golang'
-    args: ['git', 'config', '--global', 'user.name', 'Your Name']
-    dir: '/workspace'
+
+  # Fetch the Hugo binary
+  - name: 'gcr.io/cloud-builders/wget'
+    args:
+      - '--quiet'
+      - '-O'
+      - '/hugo/hugo.tar.gz'
+      - 'https://github.com/gohugoio/hugo/releases/download/v${_HUGO_VERSION}/hugo_extended_${_HUGO_VERSION}_Linux-64bit.tar.gz'
+    waitFor: ['-']
+    volumes:
+      - name: hugo
+        path: /hugo
+
+  # Delete the existing content of the public folder and regenerate the
+  # site with Hugo.
+  # This updates the path for the generated content for Hugo. This is temporary
+  # so we can start generating new content on a separate branch before switching
+  # the site over to use that branch.
+  - name: 'node:14-stretch'
+    args:
+      - 'bash'
+      - '-c'
+      - |
+        tar -C /hugo -xzf /hugo/hugo.tar.gz
+        sed -i 's/"..\/docs"/"..\/public"/g' site/config.toml
+        rm -fr public/*
+        (cd site && HUGO_ENV="production" /hugo/hugo)
+    dir: 'kpt'
+    volumes:
+      - name: hugo
+        path: /hugo
+
+  # Add the new content of the public folder
+  - name: 'gcr.io/cloud-builders/git'
+    args: ['add', '.']
+    dir: 'kpt/public'
     volumes:
       - name: home
         path: /root
-  - name: 'mirror.gcr.io/library/golang'
-    args: ['make', 'all']
-    env: ['GO111MODULE=on']
-    dir: '/workspace'
+
+  # Create a new commit referencing the source commit on the master branch
+  - name: 'gcr.io/cloud-builders/git'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+        if [ -n "$(git status --porcelain)" ]; then
+          git commit -m "Site gen from revision ${COMMIT_SHA}"
+        else
+          echo "no changes";
+        fi
+    dir: 'kpt/public'
     volumes:
-      - name: go-modules
-        path: /go
       - name: home
         path: /root
 
-  # push the container image
-  - name: 'gcr.io/cloud-builders/docker'
-    args: ['push', 'gcr.io/$PROJECT_ID/kpt:${BRANCH_NAME}']
-
-  - name: 'gcr.io/cloud-builders/docker'
-    args: ['push', 'gcr.io/kpt-dev/example-functions:${BRANCH_NAME}']
-
-  # push the binaries
-  - name: 'gcr.io/cloud-builders/gsutil'
-    args: ['cp', '-r', '-a', 'public-read', '/workspace/bin/', 'gs://kpt-dev/']
+  # Force push the changes to the gh-pages branch. We use force here to make
+  # sure that we get the changes in even if another job committed other changes
+  # in between the checkout and the commit.
+  - name: 'gcr.io/cloud-builders/git'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+        if git log -1 --pretty=%B | cat | grep -q "${COMMIT_SHA}"; then
+          git push origin gh-pages -f
+        else
+          echo "no changes, skipping push"
+        fi
+    dir: 'kpt/public'
+    volumes:
+      - name: home
+        path: /root
 
 timeout: '60m'
```

Comment:
How long does the job usually take?

Reply:
Around 25 minutes. Eventually we might consider updating the site in a separate job from building the binaries.